tests/activation/test_prompts.py

```diff
@@ -0,0 +1,261 @@
+            sys.platform.startswith("linux") and shell in ["cmd", "powershell"]
+        ):
+            pytest.skip(platform_incompat.format(shell, platform))
+
+        if sys.platform.startswith("win") and shell == "xonsh":
+            pytest.skip("Provisioning xonsh on windows is unreliable")
+
+        if shell == "xonsh" and sys.version_info < (3, 4):
+            pytest.skip("xonsh requires Python 3.4 at least")
+
+        if shell == "powershell":
+            test_ps1 = tmp_path_factory.mktemp("posh_test") / "test.ps1"
+            test_ps1.write_text("echo foo\n")
+
+            if 0 != subprocess.call("powershell -File {}".format(str(test_ps1)), shell=True):
+                pytest.skip("powershell script execution not enabled")
+                # Enable with:  PS> Set-ExecutionPolicy -scope currentuser -ExecutionPolicy Bypass -Force;
+                # Disable with: PS> Set-ExecutionPolicy -scope currentuser -ExecutionPolicy Restricted -Force;
+
+    return check
+
+
+@pytest.fixture(scope="module")
+def custom_prompt_root(tmp_path_factory):
+    """Provide Path to root with default and custom venvs created."""
+    root = tmp_path_factory.mktemp("custom_prompt")
+    virtualenv.create_environment(
+        str(root / ENV_CUSTOM), prompt=PREFIX_CUSTOM, no_setuptools=True, no_pip=True, no_wheel=True
+    )
+
+    _, _, _, bin_dir = virtualenv.path_locations(str(root / ENV_DEFAULT))
+
+    bin_dir_name = os.path.split(bin_dir)[-1]
+
+    return root, bin_dir_name
+
+
+@pytest.fixture(scope="module")
+def clean_python_root(clean_python):
+    root = Path(clean_python[0]).resolve().parent
+    bin_dir_name = os.path.split(clean_python[1])[-1]
+
+    return root, bin_dir_name
+
+
+@pytest.fixture(scope="module")
+def get_work_root(clean_python_root, custom_prompt_root):
+    def pick_root(env):
+        if env == ENV_DEFAULT:
+            return clean_python_root
+        elif env == ENV_CUSTOM:
+            return custom_prompt_root
+        else:
+            raise ValueError("Invalid test environment")
+
+    return pick_root
+
+
+@pytest.fixture(scope="module")
+def shell_info():
+    class ShellInfo:
+
+        execute_cmds = EchoDict({"cmd": "", "powershell": "powershell -File "})
+
+        # {shell}.script.(normal|suppress).(default|custom)[extension]
+        script_template = "{0}.script.{1}.{2}{3}"
+
+        # {shell}.out.(normal|suppress).(default|custom)
+        output_template = "{0}.out.{1}.{2}"
+
+        testscript_extensions = defaultdict(lambda: "")
+        testscript_extensions.update({"powershell": ".ps1", "cmd": ".bat"})
+
+        preamble_cmds = {
+            "bash": "",
+            "fish": "",
+            "csh": "set prompt=%",
+            "xonsh": "$VIRTUAL_ENV = ''; $PROMPT = '{env_name}$ '",
+            "cmd": "@echo off & set PROMPT=$P$G",
+            "powershell": "",
+        }
+
+        prompt_cmds = {
+            "bash": 'echo "$PS1"',
+            "fish": "fish_prompt; echo ' '",
```

Comment:
not the biggest fan of this dictionaries, maybe group by shell instead and use a namedtuple 🤔

Reply:
Fixed in e0f7394